### PR TITLE
solution

### PR DESCRIPTION
**Pull request by pasha1932**
[DEMO LINK](https://pasha1932.github.io/react_goods-selector-js/)

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -14,57 +15,69 @@ export const goods = [
...
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
+  return (
+    <main className="section container">
+      {selectedGood === '' ? (
```
Comment (Viktor-Kost) [inline src/App.jsx:23]:
```suggestion
      {!selectedGood ? (
```

File: src/App.jsx [hunk 2]
```diff
@@ -14,57 +15,69 @@ export const goods = [
...
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
+      <table className="table">
+        <tbody>
+          {goods.map(good => (
+            <tr
```
Comment (Viktor-Kost) [inline src/App.jsx:44]: 1. add a react `key`
2. move `selectedGood === good` to a variable
```suggestion
          {goods.map(good => {
            const isActive = selectedGood === good;
            
            <tr
              key={good}
```

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]

File: src/App.jsx (hunk 2 above)
Comment (pasha1932) [reply src/App.jsx:44]: > 1. add a react `key`
> 2. move `selectedGood === good` to a variable

image

File: src/App.jsx [hunk 3]
```diff
@@ -14,57 +15,70 @@ export const goods = [
...
+                    }}
+                  >
+                    +
+                  </button>
+                ) : (
+                  <button
+                    data-cy="RemoveButton"
+                    type="button"
+                    className="button is-info"
+                    onClick={() => {
```
Comment (anastasiiavorobiova) [inline src/App.jsx:66]: Consider creating a function for resetting a value

File: src/App.jsx (hunk 2 above)
Comment (anastasiiavorobiova) [reply src/App.jsx:44]: Feel free to ask for help in the chat

File: src/App.jsx [hunk 4]
```diff
@@ -14,57 +15,70 @@ export const goods = [
...
+          No goods selected
+        </h1>
+      ) : (
+        <h1 className="title is-flex is-align-items-center">
+          {selectedGood} is selected
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={() => {
```
Comment (anastasiiavorobiova) [inline src/App.jsx:34]: Consider reusing a function for resetting a value

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Great work! Almost done!

File: src/App.jsx [hunk 5]
```diff
@@ -14,57 +15,74 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
+  // let isActive = true;
```
Comment (volodymyr-soltys97) [inline src/App.jsx:20]: Remove comment
```suggestion
```

File: src/App.jsx [hunk 6]
```diff
@@ -14,57 +15,74 @@ export const goods = [
...
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
+  function resetValue() {
```
Comment (volodymyr-soltys97) [inline src/App.jsx:22]: Use function expression here

File: src/App.jsx [hunk 7]
```diff
@@ -14,57 +15,74 @@ export const goods = [
...
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+            return (
+              <tr
+                key={good}
+                data-cy="Good"
+                className={`${isActive ? 'has-background-success-light' : ''}`}
```
Comment (volodymyr-soltys97) [inline src/App.jsx:52]: Use the classnames library for add classes with condition

File: src/App.jsx [hunk 8]
```diff
@@ -14,57 +15,74 @@ export const goods = [
...
+              >
+                <td>
+                  {!isActive ? (
+                    <button
+                      data-cy="AddButton"
+                      type="button"
+                      className="button"
+                      onClick={() => {
+                        setSelectedGood(good);
+                      }}
```
Comment (volodymyr-soltys97) [inline src/App.jsx:62]:
```suggestion
                      onClick={() => setSelectedGood(good)}
```

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Almost done!
Let's make your code better

**Review by lerastarynets — APPROVED** [review]
You're doing amazing, keep it up!

File: src/App.jsx (hunk 2 above)
Comment (Viktor-Kost) [reply src/App.jsx]: @pasha1932 hello!
Sorry, it was my mistake. Forgot to add `return`
It should looks like 
```
          {goods.map(good => {
            const isActive = selectedGood === good;
            
            return (
              <tr
                key={good}
```

**Pull request closed without merging**